zerver/views/previewable.py

```diff
@@ -0,0 +1,34 @@
+import re
+from urllib.parse import urlparse
+
+from django.http import HttpRequest, HttpResponse
+from django.utils.translation import gettext as _
+
+from zerver.lib.exceptions import JsonableError
+from zerver.lib.github import get_issue_or_pr_data
+from zerver.lib.request import REQ, has_request_variables
+from zerver.lib.response import json_success
+from zerver.models import UserProfile
+
+
+@has_request_variables
+def get_previewable_data(
+    request: HttpRequest, user_profile: UserProfile, url: str = REQ()
+) -> HttpResponse:
+    parsed_url = urlparse(url)
+    if (
+        parsed_url.hostname
+        and re.match(r"^(?:www\.)?github\.com$", parsed_url.hostname) is not None
+    ):
+        path_split = parsed_url.path.split("/")
+        if path_split[3] in ["issues", "pull"]:
+            data = get_issue_or_pr_data(
+                owner=path_split[1],
+                repo=path_split[2],
+                issue_number=path_split[4],
+            )
+            if data["result"] == "success":
+                return json_success(request, data)
+            error_msg = data["msg"]
+            raise JsonableError(f"{_(error_msg)}")
```

Comment:
This isn't translatable, since it can be an arbitrary string.  The f-string isn't helping improve the readability, either. :)

I'm guessing that you want to cache the failure, which is why `get_issue_or_pr_data` isn't raising the JsonableError directly.  Which is reasonable for 404s -- those are unlikely to want to try to fetch fresh each time. If you're going to do this, I think you need to return some opaque constant from get_issue_or_pr_data, and check those values here and raise JsonableError with the user-facing string explicitly.